dotCMS/src/main/java/com/dotcms/publisher/ajax/RemotePublishAjaxAction.java

```diff
@@ -713,19 +713,18 @@ public void uploadBundle ( HttpServletRequest request, HttpServletResponse respo
 
         FileItemFactory factory = new DiskFileItemFactory();
         ServletFileUpload upload = new ServletFileUpload( factory );
-        @SuppressWarnings ("unchecked")
-        List<FileItem> items = (List<FileItem>) upload.parseRequest( request );
-
-        InputStream bundle = items.get( 0 ).getInputStream();
-        String bundleName = items.get( 0 ).getName();
-        String bundlePath = ConfigUtils.getBundlePath() + File.separator;
-        String bundleFolder = bundleName.substring( 0, bundleName.indexOf( ".tar.gz" ) );
-        String endpointId = getUser().getUserId();
-        response.setContentType( "text/html; charset=utf-8" );
         PrintWriter out = response.getWriter();
-
         PublishAuditStatus status;
+        String bundleName = null;
         try {
+            @SuppressWarnings ("unchecked")
+            List<FileItem> items = (List<FileItem>) upload.parseRequest( request );
+            InputStream bundle = items.get( 0 ).getInputStream();
+            bundleName = sanitizeBundleName(items.get(0).getName());
+            String bundlePath = ConfigUtils.getBundlePath() + File.separator;
+            String bundleFolder = bundleName.substring( 0, bundleName.indexOf( ".tar.gz" ) );
+            String endpointId = getUser().getUserId();
+            response.setContentType( "text/html; charset=utf-8" );
```

Comment:
it wouldn't a bad idea to set this as parameter, eventually someone might need a diff charset